packages/debugger/lib/controller/sagas/index.js

```diff
@@ -207,8 +207,9 @@ function* stepOver() {
     // line (which may be in a new file)
     (currentDepth > startingDepth ||
       (currentLocation.source.id === startingLocation.source.id &&
-        currentLocation.sourceRange.lines.start.line ===
-          startingLocation.sourceRange.lines.start.line))
+        startingLocation.isMultiline ^
+          (currentLocation.sourceRange.lines.start.line ===
+            startingLocation.sourceRange.lines.start.line)))
```

Comment:
whoa! Can you explain this logic @dongmingh? Is there a more readable way to express this?

Reply:
1. this is the condition in a do-while loop

2. there are two cases: a multi line and a single line.  The existing code fails the multi line case.
  - For multi line case:
     - isMultiline is true
     - the starting lines and ending lines are the same at the starting and ending of the source range, so continue stepping until the starting lines are the same
    ```
       startingLocation.isMultiline && 
          (currentLocation.sourceRange.lines.start.line !==
            startingLocation.sourceRange.lines.start.line)
    ```
  - For single line case:
     - isMultiline is false
     - continue stepping until the starting lines are different
    ```
       !startingLocation.isMultiline && 
          (currentLocation.sourceRange.lines.start.line ===
            startingLocation.sourceRange.lines.start.line)
    ```

3. if combine the two logics above into one statement, see below, which is very `long` and hard to read.
   ```
        (startingLocation.isMultiline && 
          (currentLocation.sourceRange.lines.start.line !==
            startingLocation.sourceRange.lines.start.line))  ||
        (!startingLocation.isMultiline && 
          (currentLocation.sourceRange.lines.start.line ===
            startingLocation.sourceRange.lines.start.line))
   ```

4. the statement can be simplified as
   ```
        startingLocation.isMultiline ^
          (currentLocation.sourceRange.lines.start.line ===
            startingLocation.sourceRange.lines.start.line)
   ```

It may help users if adding detail comments here for explanation.

Reply:
Thanks @dongmingh. The comments preceding the `while` block seem outdated with the new code and if we combining bitwise and logical operators it'd be helpful to be explicit in the description/comment.

Reply:
@dongmingh `a ^ b` is equivalent to `a !== b` for booleans and avoids the casting and between boolean and number for the bitwise operator.

```
        startingLocation.isMultiline !==
          (currentLocation.sourceRange.lines.start.line ===
            startingLocation.sourceRange.lines.start.line)))
```

Reply:
very nice, thanks